genai/image_generation/imggen_canny_ctrl_type_with_txt_img.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def canny_edge_customization(output_gcs_uri: str) -> str:
+    # [START googlegenaisdk_imggen_canny_ctrl_type_with_txt_img]
+    from google import genai
+    from google.genai.types import ControlReferenceConfig, ControlReferenceImage, EditImageConfig, Image
+
+    client = genai.Client()
+
+    # TODO(developer): Update and un-comment below line
+    # output_gcs_uri = "gs://your-bucket/your-prefix"
+
+    # Create a reference image out of an existing canny edge image signal
```

Comment:
If possible I would like to have a http link for input that any logging gmail user can check

```
    # Create a reference image by using below canny edge image signal
    # using https://storage.cloud.google.com/cloud-samples-data/generative-ai/image/car_canny.png as reference
```